static/game/bullet.js

```diff
@@ -131,3 +144,137 @@ export class Bullet extends Entity {
     }
   }
 }
+
+export class Beam extends Bullet {
+  /**
+   * Constructs a new beam. Beams don't have velocity or acceleration. Instead,
+   * they have a starting point (pos) and length
+   * @param {Vector} [pos] the position of the origin of the beam
+   * @param {Vector} [dir] the direction the beam is facing
+   * @param {Vector} [acc] ignored
```

Comment:
why include acceleration as an optimal parameter if it's ignored? that means in order to set an owner you would always need to give it an undefined parameter.
Also, optional parameters should be after non-optional ones.

Reply:
It's to match the constructor signature of `Bullet`, so when you can do `new [Bullet, Beam][Math.floor(Math.random() * 2)]()` and without having to change the arguments you pass to it

Reply:
I think that this is more gross than just having an if statement when shooting a beam or bullet

Reply:
i mean it's cool that js can do that but this isn't the use case for it

Reply:
It makes it easier if we add other types of bullets (e.g. orbitals) to have them all have the same constructor arguments. I'd argue that it's better to have as few arguments as possible and just set them later, like this:
```
const b = new Bullet();
b.pos = whatever;
b.vel = whatever;
b.acc = whatever;
```
rather than
```
const b = new Bullet(pos, vel, acc);
```

Reply:
We should have a wrapper class "Shootable" that has arguments in a better order then

Reply:
I'm having Java flashbacks :fearful:

Reply:
Also, it's good to have things as arguments to the constructor rather than defined later if they're non-optional and don't have a reasonable default.
bullets don't have reasonable defaults for any of those (they should be different per use case), so they should be in the constructor.

Reply:
> It's to match the constructor signature of `Bullet`, so when you can do `new [Bullet, Beam][Math.floor(Math.random() * 2)]()` and without having to change the arguments you pass to it

If the reason is to match the constructor signature then I'm actually okay with that.

Reply:
Still don't  like this but ok